### PR TITLE
[Chore] Moves Drag and Drop to new branch

### PR DESCRIPTION
**Pull request by ashwin-pc**
### Description
The old `feature/d-and-d-379` branch had a merge commit that prevented it from being rebased off the latest main. This PR cherry picks those commits onto the latest commit on main 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [x] `yarn test:jest`
    - [x] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/wizard/public/application/components/data_tab/config_panel.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.wizConfigPanel {
+  background: #f0f1f3;
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/components/data_tab/config_panel.scss:2]: Should this come from the theme?

File: src/plugins/wizard/public/application/components/data_tab/config_section.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  & .euiFormRow__labelWrapper {
+    margin-left: $euiSizeS;
+  }
+}
+
+.wizConfigSection__dropTarget {
+  @include euiSlightShadow;
+  background: $euiColorEmptyShade;
+  border: $euiBorderThin;
+  box-shadow: 0px 2px 2px rgba(152, 162, 179, 0.15);
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/components/data_tab/config_section.scss:19]: Can this be pulled from the theme?

File: src/plugins/wizard/public/application/components/data_tab/config_panel.scss (hunk 1 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/data_tab/config_panel.scss:2]: We currently dont have this value (or similar) in EUI. The UI is still being updated. Will address all hardcoded colors in the fit and finish issue when we go over the styles for the whole app. We will either have to add these values in EUI or update our styles to follow EUI.

File: src/plugins/wizard/public/application/components/data_tab/config_section.scss (hunk 2 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/data_tab/config_section.scss:19]: same as above.

File: src/plugins/wizard/public/application/components/data_tab/field_selector_field.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.wizFieldSelectorField {
+  @include euiBottomShadowSmall;
+  padding: $euiSizeXS;
+  background-color: $euiColorEmptyShade;
+  border: $euiBorderThin;
+  margin-top: $euiSizeS;
+
+  & > button {
+    align-items: center;
+    gap: 4px;
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/components/data_tab/field_selector_field.scss:10]: Can this come from the theme too?

File: src/plugins/wizard/public/application/components/data_tab/field_selector_field.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/components/data_tab/field_selector_field.tsx:7]: Wrong header

File: src/plugins/wizard/public/application/components/data_tab/field_selector_field.scss (hunk 3 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/data_tab/field_selector_field.scss:10]: The style for this is yet to be updated. @btzeng is still working on the final style for this section

File: src/plugins/wizard/public/application/components/workspace.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      </EuiFlexGroup>
+      <EuiPanel className="wizCanvas">
+        {children ? (
+          children
+        ) : (
+          <EuiFlexItem className="wizWorkspace__empty">
+            <EuiEmptyPrompt
+              iconType="visBarVertical"
+              title={<h2>Welcome to the wizard!</h2>}
+              body={<p>Drag some fields onto the panel to visualize some data.</p>}
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/components/workspace.tsx:41]: Can we have `i18n.translate` for the `title` and `body`?

Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/workspace.tsx:41]: We have a separate issue to track updating all the hard coded string that we ported over from the POC #958

File: src/plugins/wizard/public/application/utils/get_top_nav_config.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/utils/get_top_nav_config.tsx:7]: Wrong header

File: src/plugins/wizard/public/application/components/data_tab/field_selector_field.tsx (hunk 4 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/data_tab/field_selector_field.tsx:7]: Good catch, will update

File: src/plugins/wizard/public/application/utils/get_top_nav_config.tsx (hunk 6 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/utils/get_top_nav_config.tsx:7]: Good catch, will update

File: src/plugins/wizard/public/application/utils/state_management/config_slice.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+// TODO: Temp. Remove once visualizations can be refgistered and editor configs can be passed along
+// TODO: this is a placeholder while the config section is iorned out
+const initialState: ConfigState = {
+  configSections: {
+    x: {
+      title: 'X Axis',
+      fields: [],
+    },
+    y: {
+      title: 'Y Axis',
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/utils/state_management/config_slice.ts:28]: Should we have `i18n.translate` for the 2 `title` props?

File: src/plugins/wizard/public/services/type_service/type_service.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/services/type_service/type_service.ts:7]: Wrong header

File: src/plugins/wizard/public/application/utils/state_management/config_slice.ts (hunk 7 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/utils/state_management/config_slice.ts:28]: Yes, but this code will be completely replaced by the Contributions PR that is following this. There it is set by the visualization type directly. That being said, all this code is still in flux. Thats why we have a separate issue to track updating all hard coded strings.

File: src/plugins/wizard/public/visualizations/bar_chart/index.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+import { VisualizationTypeOptions } from '../../services/type_service/visualization_type';
+
+export const createBarChartConfig = (): VisualizationTypeOptions => {
+  return {
+    name: 'bar_chart',
+    title: 'Bar Chart',
+    icon: 'visBarVertical',
+    description: 'This is a bar chart',
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/visualizations/bar_chart/index.ts:13]: Is it important to localize these?

**Review by AMoo-Miki — APPROVED** [review]
Left some notes.

File: src/plugins/wizard/public/visualizations/bar_chart/index.ts (hunk 9 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/visualizations/bar_chart/index.ts:13]: Yep. Once I finalize the API here, ill update the strings in #958

File: src/plugins/wizard/public/services/type_service/type_service.ts (hunk 8 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/services/type_service/type_service.ts:7]: Good catch, will update.

**Pull request merged**